### PR TITLE
Add STS support

### PR DESCRIPTION
**Pull request by msakrejda**
@cyberdelia or @fdr can you take a look? I'm bumbling through the ruins of my Python knowledge

### CODE REVIEW

File: tablesnap [hunk 1]
```diff
@@ -56,7 +56,8 @@ default_chunk_size = 256
 default_listen_event = 'IN_MOVED_TO'
 
 class UploadHandler(pyinotify.ProcessEvent):
-    def my_init(self, threads=None, key=None, secret=None, bucket_name=None,
+    def my_init(self, threads=None, key=None, secret=None, token=None,
```
Comment (cyberdelia) [inline tablesnap:59]: `my_init` O_o

**Comment by cyberdelia** [comment]
Looks good.

File: tablesnap (hunk 1 above)
Comment (msakrejda) [inline tablesnap:59]: Right? Do you know what this is? It doesn't seem to be called explicitly anywhere. Is it some implicit thing?

Comment (cyberdelia) [inline tablesnap:59]: Apparently, it's a hook for `pyinotify.ProcessEvent`.

Comment (msakrejda) [inline tablesnap:59]: Huh. Well, when in Rome...

**Comment by fdr** [comment]
Looks superficially fine to me as well.

**Pull request merged**